src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/CaseEventHandlerController.java

```diff
@@ -53,4 +61,36 @@ public CaseEventHandlerController(List<CaseEventHandler> handlerServices) {
 
     }
 
+    @ApiOperation("Handles the CCD case event message")
+    @ApiResponses({
+        @ApiResponse(
+            code = 201,
+            message = "Message processed successfully",
+            response = Object.class)
+    })
+    @PostMapping(path = "/messages/{message_id}", consumes = {MediaType.APPLICATION_JSON_VALUE})
+    @SuppressWarnings("PMD.DataflowAnomalyAnalysis")
+    @ResponseStatus(HttpStatus.CREATED)
+    public CaseEventMessage caseEventHandler(@Valid @RequestBody String message,
+                                             @PathVariable("message_id") final String messageId,
+                                             @RequestParam(value = "from_dlq",
+                                                 required = false) final Boolean fromDlq) {
+        if (fromDlq != null && fromDlq) {
```

Comment:
Can we add test to fully cover this if condition?

Reply:
Added a unit test for this.